framework/auth/views.py

```diff
@@ -230,9 +239,16 @@ def send_confirm_email(user, email):
     except NoResultsFound:
         merge_target = None
 
+    campaign = campaigns.campaign_for_user(user)
+    # Choose the appropriate email template to use
+    if merge_target:
+        mail_template = mails.CONFIRM_MERGE
+    else:
```

Comment:
That's a good point. A change is less pressing in that case.

Speaking generally, though, I'm still a bit leery of the generic else condition here, particularly since we're still evolving the campaigns feature. 

The behavior of that other function is the only thing keeping this view from doing awkward things, and we're not 100% committed to the design of the other function. By making this function responsible for checking its own assumptions, you can avoid coupling it to pieces whose behavior may depend later.